src/main/java/org/cascadebot/cascadebot/utils/buttons/ButtonGroup.java

```diff
@@ -18,18 +18,45 @@
 
     private List<Button> buttons;
     private long ownerId;
+    private long channelId;
     private long guildId;
 
-    private long messageId;
+    private long messageId = 0;
 
-    public ButtonGroup(long ownerId, long guildId) {
+    public ButtonGroup(long ownerId, long channelId, long guildId) {
         buttons = new ArrayList<>();
         this.ownerId = ownerId;
+        this.channelId = channelId;
         this.guildId = guildId;
     }
 
     public void addButton(Button button) {
         buttons.add(button);
+        if (messageId != 0) {
+            CascadeBot.INS.getShardManager().getGuildById(guildId).getTextChannelById(channelId).getMessageById(messageId).queue(button::addReaction);
+        }
+    }
+
+    public void removeButton(Button button) {
+        buttons.remove(button);
+        if (messageId != 0) {
+            CascadeBot.INS.getShardManager().getGuildById(guildId).getTextChannelById(channelId).getMessageById(messageId).queue(message -> {
+                for (MessageReaction reaction : message.getReactions()) {
+                    MessageReaction.ReactionEmote reactionEmote = reaction.getReactionEmote();
+                    if (button instanceof Button.UnicodeButton && !reactionEmote.isEmote()) {
+                        Button.UnicodeButton unicodeButton = (Button.UnicodeButton) button;
+                        if (reactionEmote.getName().equals(unicodeButton.unicode)) {
+                            reaction.removeReaction(CascadeBot.INS.getSelfUser()).queue();
```

Comment:
Don't you want to remove all reactions, not just the one from the bot?

Reply:
The other reactions should have already been removed because buttons

Reply:
"because buttons"? Where do you remove all the other reactions?

Reply:
In the button even listener. their should only ever be the bots reaction on buttons due to them being removed due to the buttons.

Reply:
Oh I get you, perhaps we should just remove all anyway? It will eliminate the chance of a wierd bug happening

Reply:
could do, but it would cause more api request and we might get rate limited.

Reply:
It's actually just one API request thankfully [see here](https://discordapp.com/developers/docs/resources/channel#delete-all-reactions)

Reply:
You that delete ALL reactions, not reactions of a specific type.

Reply:
Oof